x-pack/plugins/stack_alerts/common/build_sorted_events_query.test.ts

```diff
@@ -0,0 +1,347 @@
+    query.searchAfterSortId = '';
+    expect(buildSortedEventsQuery(query)).toEqual({
+      allowNoIndices: true,
+      index: ['index-name'],
+      size: 100,
+      ignoreUnavailable: true,
+      track_total_hits: true,
+      body: {
+        docvalue_fields: [
+          {
+            field: 'timefield',
+            format: 'strict_date_optional_time',
+          },
+        ],
+        query: {
+          bool: {
+            filter: [
+              {},
+              {
+                bool: {
+                  filter: [
+                    {
+                      range: {
+                        timefield: {
+                          gte: '2021-01-01T00:00:10.123Z',
+                          lte: '2021-01-23T12:00:50.321Z',
+                          format: 'strict_date_optional_time',
+                        },
+                      },
+                    },
+                  ],
+                },
+              },
+              {
+                match_all: {},
+              },
+            ],
+          },
+        },
+        sort: [
+          {
+            timefield: {
+              order: 'asc',
+            },
+          },
+        ],
+      },
+    });
+  });
+
+  test('it includes searchAfterSortId if it is a valid string', () => {
+    const sortId = '123456789012';
+    query.searchAfterSortId = sortId;
+    expect(buildSortedEventsQuery(query)).toEqual({
+      allowNoIndices: true,
+      index: ['index-name'],
+      size: 100,
+      ignoreUnavailable: true,
+      track_total_hits: true,
+      body: {
+        docvalue_fields: [
+          {
+            field: 'timefield',
+            format: 'strict_date_optional_time',
+          },
+        ],
+        query: {
+          bool: {
+            filter: [
+              {},
+              {
+                bool: {
+                  filter: [
+                    {
+                      range: {
+                        timefield: {
+                          gte: '2021-01-01T00:00:10.123Z',
+                          lte: '2021-01-23T12:00:50.321Z',
+                          format: 'strict_date_optional_time',
+                        },
+                      },
+                    },
+                  ],
+                },
+              },
+              {
+                match_all: {},
+              },
+            ],
+          },
+        },
+        sort: [
+          {
+            timefield: {
+              order: 'asc',
+            },
+          },
+        ],
+        search_after: [sortId],
+      },
+    });
+  });
+
+  test('it includes searchAfterSortId if it is a valid number', () => {
+    const sortId = 123456789012;
+    query.searchAfterSortId = sortId;
+    expect(buildSortedEventsQuery(query)).toEqual({
+      allowNoIndices: true,
+      index: ['index-name'],
+      size: 100,
+      ignoreUnavailable: true,
+      track_total_hits: true,
```

Comment:
Is it worth making `track_total_hits ` configurable? 🤔 
In a large data set setting this to true will impact performance, and I can see situations where I care more about knowing that something is happening than knowing all the instances in which it happens.

I haven't thought about this in depth, so I'm likely missing something, but I'm not 100% sure why we need to specify this.